packages/@aws-cdk/aws-cognito/test/user-pool-attr.test.ts

```diff
@@ -178,4 +179,113 @@ describe('User Pool Attributes', () => {
+
+      // WHEN
+      const clientAttributes = ClientAttributes.empty().withStandardAttributes({
+        address: true,
+        birthdate: true,
+        email: true,
+        emailVerified: true,
+        familyName: true,
+        fullname: true,
+        gender: true,
+        givenName: true,
+        lastUpdateTime: true,
+        locale: true,
+        middleName: true,
+        nickname: true,
+        phoneNumber: true,
+        phoneNumberVerified: true,
+        preferredUsername: true,
+        profilePage: true,
+        profilePicture: true,
+        timezone: true,
+        website: true,
+      }).withCustomAttributes(...customAttributes);
+      const attributes = clientAttributes.attributes();
+
+      // THEN
+      expect(attributes.length).toEqual(20);
+      expect(attributes).toContain('preferred_username');
+      expect(attributes).toContain('email_verified');
+      expect(attributes).toContain('phone_number_verified');
+      expect(attributes).toContain('custom:my_attribute');
+    });
+
+    test('create ClientAttributes copying another one', () => {
+      // GIVEN
+      const original = ClientAttributes.empty()
+        .withStandardAttributes({ email: true })
+        .withCustomAttributes('custom1');
+      const copied = original
+        .withStandardAttributes({ emailVerified: true })
+        .withCustomAttributes('custom2');
+
+      // WHEN
+      const originalAttributes = original.attributes();
+      const copiedAttributes = copied.attributes();
+
+      // THEN
+      expect(originalAttributes.length).toEqual(2);
+      expect(copiedAttributes.length).toEqual(4);
+      // originals MUST NOT contain the added ones
+      expect(originalAttributes).toContain('email');
+      expect(originalAttributes).toContain('custom:custom1');
+      expect(originalAttributes).not.toContain('email_verified');
+      expect(originalAttributes).not.toContain('custom:custom2');
+      // copied MUST contain all attributes
+      expect(copiedAttributes).toContain('email');
+      expect(copiedAttributes).toContain('custom:custom1');
+      expect(copiedAttributes).toContain('email_verified');
+      expect(copiedAttributes).toContain('custom:custom2');
+    });
+
+    test('create ClientAttributes with custom attributes only', () => {
+      // GIVEN
+      const customAttributes = ['custom:my_first', 'my_second'];
+
+      // WHEN
+      const clientAttributes = ClientAttributes.empty().withCustomAttributes(...customAttributes);
+      const attributes = clientAttributes.attributes();
+
+      // EXPECT
+      expect(attributes.length).toEqual(2);
+      expect(attributes).toContain('custom:my_first');
+      expect(attributes).toContain('custom:my_second');
+    });
+
+    test('create ClientAttributes with all StandardAttributeNames', () => {
+      // this test is intended to check if changes in the `StandardAttributeNames` constant
+      // does not reflect as changes in the `StandardAttributesMask`
```

Comment:
Nice! Can you update this test to also do the reverse? i.e., changes in StandardAttributesMask are reflected in StandardAttributeNames

Reply:
The problem with this is that `StandardAttributesMask` is an interface: this means the interface is erased at runtime and I have no idea how to iterate on its properties and detect changes. The only way I see is parsing directly the `lib/user-pool-attr.ts` or the `lib/user-pool-attr.d.ts`. Do you have any hint on how to proceed?

On the other end, the problem arise only if there is a new property added to the `StandardAttributesMask` but no test is setup to check if it is added to the `(new ClientAttributes).attributes()` array.